### DISCUSSION TITLE
How to prevent cycles in DAG graph?

### DISCUSSION THREAD

**Discussion by nikzanda** [antvis/XFlow#317] [Q&A]
My goal is to have a graph with one or more initial vertex and _the nodes cannot be connected to the previous ones_.

Here are some screenshots showing what I want. I have circled the starting edge in blue, the green tick is where node can connect and the red x is where node should not connect.
![1](https://user-images.githubusercontent.com/71180446/188395721-f5beb25f-b535-4c1e-9f03-ab7b51e8b0e2.png)
![2](https://user-images.githubusercontent.com/71180446/188395724-914e50ff-2b96-448a-8097-4b7cf9bbccad.png)
![3](https://user-images.githubusercontent.com/71180446/188395728-be47b8d6-99cb-48f7-b8a9-0a38ea416cfd.png)

Also your example not working:
![4](https://user-images.githubusercontent.com/71180446/188397838-41e25aa6-2dda-4a1c-a4eb-0a13eef2285c.png)

**Comment by bliberi** [comment]
I would like to do the same thing, has anyone achieved to prevent this?
